torch/ao/quantization/_quantize_pt2e.py

```diff
@@ -53,27 +47,32 @@ def prepare_pt2e_quantizer(
     model: GraphModule,
     quantizer: Quantizer,
 ):
-    # TODO: move this information to fx node itself
-    node_name_to_scope: Dict[str, Tuple[str, type]] = {}
-    for n in model.graph.nodes:
-        nn_module_stack = n.meta.get("nn_module_stack", None)
-        current_scope = ("", type(None))
-        if nn_module_stack:
-            bt = list(nn_module_stack.values())[-1]
-            current_scope = (bt[0].split(".")[-1], bt[1])
-        node_name_to_scope[n.name] = current_scope
-
+    node_name_to_scope = _build_node_name_to_scope(model)
     # TODO: check qconfig_mapping to make sure conv and bn are both configured
     # to be quantized before fusion
     # TODO: (maybe) rewrite this with subgraph_rewriter
     _fuse_conv_bn_(model)
-    model = prepare(
-        model,
-        quantizer,
-        False,  # is_qat
-        node_name_to_scope,
-    )
+    quantizer.annotate(model)
+    quantizer.validate(model)
+    model = prepare(model, node_name_to_scope, is_qat=False)
+    # TODO: remove hack when we have better support for pattern matching
+    # move around the observer for addmm
+    _rearrange_weight_observer_for_decomposed_linear(model)
```

Comment:
don't need to do it in this PR but I think we can probably remove this now

Reply:
This is also something I would like to not have to worry about. So we either pattern match the subgraph or use module level info